content/blog/2022/10/envvar-policy.md

```diff
@@ -0,0 +1,138 @@
+and all the  artifacts, including the BOM files, are signed with Sigstore
+
+### What is so useful about the new environment-variable policy?
+
+The recent added policy https://github.com/kubewarden/env-variable-secrets-scanner-policy allow users
+to check if there are some leaked sensitive data in the environment variables. However, until the
+environment-variable-policy there were no way to check other environment variables aspects.
+This is fixed now. In this new policy users can validate which variables name and values their
+resources should have. For that, the policy gives 4 rule operator to control the
+environment variable defined in the containers. They are:
+
+- `anyIn`: Works as a deny list of your usage of volumes. It checks if any of the `environmentVariables` are in the Pod/Workload resource:
+    ```yaml
+settings:
+  rules:
+    - reject: anyIn
+      environmentVariables:
+        - name: "envvar1"
+        - name: "envvar2"
+    ```
+- `anyNotIn`: Works as an allow list. Checks if any of the `environmentVariables` are not in the Pod/Workload resource:
+    ```yaml
+settings:
+  rules:
+    - reject: anyNotIn
+      environmentVariables:
+        - name: "envvar2"
+          value: "envvar2_value"
+        - name: "envvar3"
+    ```
+- `allAreUsed`: The container cannot use all listed volumes at once. Checks if all of the `environmentVariables` are in the Pod/Workload resource:
+    ```yaml
+settings:
+  rules:
+    - reject: allAreUsed
+      environmentVariables:
+        - name: "envvar2"
+          value: ""
+    ```
+
+- `notAllAreUsed`: The container can use all listed volumes at once, but only one of them. Checks if all of the `environmentVariables` are not in the Pod/Workload resource:
+```yaml
+settings:
+  rules:
+    - reject: notAllAreUsed
+      environmentVariables:
+        - name: "envvar3"
+          value: "envvar3_value"
+        - name: "envvar4"
+          value: "envvar4_value"
+    ```
+
+The user is allow to define multiple rules in the policy settings. The resource must be approved by all of them in order to be accepted into the cluster.
+
+### An example
+
+Let's see how users can enforce that their resource have all minium set of required environment variable. For that, it possible to use the `allAreUsed` operator:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: policies.kubewarden.io/v1
+kind: ClusterAdmissionPolicy
+metadata:
+  name: environment-variable-policy
+spec:
+  module: ghcr.io/kubewarden/policies/environment-variable-policy:latest
+  mutating: false
+  rules:
+  - apiGroups: [""]
+    apiVersions: ["v1"]
+    resources: ["pods", "deployments", "replicasets", "daemonsets", "replicationcontrollers", "jobs", "cronjobs"]
+    operations:
+    - CREATE
+    - UPDATE
+	settings:
+	  rules:
+	    - reject: anyIn
+	      environmentVariables:
+	        - name: "myenvvar"
+                  value: "myvalue"
+	        - name: "myenvvar2"
```

Comment:
on review, it displays as tabs, instead of spaces. Maybe is worth it to use spaces, in case people copy-paste and have a different tab setting or something.

Reply:
copy and paste doesn't work. This is the error I see:
```
error: error parsing STDIN: error converting YAML to JSON: yaml: line 15: found a tab character that violates indentation
```